### ISSUE TITLE
Most `rclone`-level Prometheus metrics are stuck at 0

### ISSUE THREAD

**Issue by apetresc** [rclone/rclone#4341]
#### What is the problem you are having with rclone?

Prometheus metrics in the `rclone_` namespace returned by `/metrics` don't seem to work uniformly. Most appear to be stuck at 0; the only one that doesn't is `rclone_errors_total`, which appears correct.

As an example, after having started up an `rcd` instance and sent it a few jobs, this is the output of `/metrics`:
```
# HELP go_gc_duration_seconds A summary of the pause duration of garbage collection cycles.
# TYPE go_gc_duration_seconds summary
go_gc_duration_seconds{quantile="0"} 8.788e-06
go_gc_duration_seconds{quantile="0.25"} 1.6143e-05
go_gc_duration_seconds{quantile="0.5"} 2.3161e-05
go_gc_duration_seconds{quantile="0.75"} 4.8196e-05
go_gc_duration_seconds{quantile="1"} 0.001190951
go_gc_duration_seconds_sum 0.166076085
go_gc_duration_seconds_count 1520
# HELP go_goroutines Number of goroutines that currently exist.
# TYPE go_goroutines gauge
go_goroutines 148
# HELP go_info Information about the Go environment.
# TYPE go_info gauge
go_info{version="go1.14.4"} 1
# HELP go_memstats_alloc_bytes Number of bytes allocated and still in use.
# TYPE go_memstats_alloc_bytes gauge
go_memstats_alloc_bytes 1.21247408e+08
# HELP go_memstats_alloc_bytes_total Total number of bytes allocated, even if freed.
# TYPE go_memstats_alloc_bytes_total counter
go_memstats_alloc_bytes_total 3.1019975136e+10
# HELP go_memstats_buck_hash_sys_bytes Number of bytes used by the profiling bucket hash table.
# TYPE go_memstats_buck_hash_sys_bytes gauge
go_memstats_buck_hash_sys_bytes 1.757951e+06
# HELP go_memstats_frees_total Total number of frees.
# TYPE go_memstats_frees_total counter
go_memstats_frees_total 2.47198317e+08
# HELP go_memstats_gc_cpu_fraction The fraction of this program's available CPU time used by the GC since the program started.
# TYPE go_memstats_gc_cpu_fraction gauge
go_memstats_gc_cpu_fraction 1.2703193400251553e-05
# HELP go_memstats_gc_sys_bytes Number of bytes used for garbage collection system metadata.
# TYPE go_memstats_gc_sys_bytes gauge
go_memstats_gc_sys_bytes 1.4186792e+07
# HELP go_memstats_heap_alloc_bytes Number of heap bytes allocated and still in use.
# TYPE go_memstats_heap_alloc_bytes gauge
go_memstats_heap_alloc_bytes 1.21247408e+08
# HELP go_memstats_heap_idle_bytes Number of heap bytes waiting to be used.
# TYPE go_memstats_heap_idle_bytes gauge
go_memstats_heap_idle_bytes 2.05185024e+08
# HELP go_memstats_heap_inuse_bytes Number of heap bytes that are in use.
# TYPE go_memstats_heap_inuse_bytes gauge
go_memstats_heap_inuse_bytes 1.28622592e+08
# HELP go_memstats_heap_objects Number of allocated objects.
# TYPE go_memstats_heap_objects gauge
go_memstats_heap_objects 145790
# HELP go_memstats_heap_released_bytes Number of heap bytes released to OS.
# TYPE go_memstats_heap_released_bytes gauge
go_memstats_heap_released_bytes 1.19472128e+08
# HELP go_memstats_heap_sys_bytes Number of heap bytes obtained from system.
# TYPE go_memstats_heap_sys_bytes gauge
go_memstats_heap_sys_bytes 3.33807616e+08
# HELP go_memstats_last_gc_time_seconds Number of seconds since 1970 of last garbage collection.
# TYPE go_memstats_last_gc_time_seconds gauge
go_memstats_last_gc_time_seconds 1.5919813280591905e+09
# HELP go_memstats_lookups_total Total number of pointer lookups.
# TYPE go_memstats_lookups_total counter
go_memstats_lookups_total 0
# HELP go_memstats_mallocs_total Total number of mallocs.
# TYPE go_memstats_mallocs_total counter
go_memstats_mallocs_total 2.47344107e+08
# HELP go_memstats_mcache_inuse_bytes Number of bytes in use by mcache structures.
# TYPE go_memstats_mcache_inuse_bytes gauge
go_memstats_mcache_inuse_bytes 6944
# HELP go_memstats_mcache_sys_bytes Number of bytes used for mcache structures obtained from system.
# TYPE go_memstats_mcache_sys_bytes gauge
go_memstats_mcache_sys_bytes 16384
# HELP go_memstats_mspan_inuse_bytes Number of bytes in use by mspan structures.
# TYPE go_memstats_mspan_inuse_bytes gauge
go_memstats_mspan_inuse_bytes 365024
# HELP go_memstats_mspan_sys_bytes Number of bytes used for mspan structures obtained from system.
# TYPE go_memstats_mspan_sys_bytes gauge
go_memstats_mspan_sys_bytes 1.654784e+06
# HELP go_memstats_next_gc_bytes Number of heap bytes when next garbage collection will take place.
# TYPE go_memstats_next_gc_bytes gauge
go_memstats_next_gc_bytes 2.2138504e+08
# HELP go_memstats_other_sys_bytes Number of bytes used for other system allocations.
# TYPE go_memstats_other_sys_bytes gauge
go_memstats_other_sys_bytes 1.316089e+06
# HELP go_memstats_stack_inuse_bytes Number of bytes in use by the stack allocator.
# TYPE go_memstats_stack_inuse_bytes gauge
go_memstats_stack_inuse_bytes 1.736704e+06
# HELP go_memstats_stack_sys_bytes Number of bytes obtained from system for stack allocator.
# TYPE go_memstats_stack_sys_bytes gauge
go_memstats_stack_sys_bytes 1.736704e+06
# HELP go_memstats_sys_bytes Number of bytes obtained from system.
# TYPE go_memstats_sys_bytes gauge
go_memstats_sys_bytes 3.5447632e+08
# HELP go_threads Number of OS threads created.
# TYPE go_threads gauge
go_threads 20
# HELP process_cpu_seconds_total Total user and system CPU time spent in seconds.
# TYPE process_cpu_seconds_total counter
process_cpu_seconds_total 11406
# HELP process_max_fds Maximum number of open file descriptors.
# TYPE process_max_fds gauge
process_max_fds 1.048576e+06
# HELP process_open_fds Number of open file descriptors.
# TYPE process_open_fds gauge
process_open_fds 20
# HELP process_resident_memory_bytes Resident memory size in bytes.
# TYPE process_resident_memory_bytes gauge
process_resident_memory_bytes 2.4389632e+08
# HELP process_start_time_seconds Start time of the process since unix epoch in seconds.
# TYPE process_start_time_seconds gauge
process_start_time_seconds 1.5918276681e+09
# HELP process_virtual_memory_bytes Virtual memory size in bytes.
# TYPE process_virtual_memory_bytes gauge
process_virtual_memory_bytes 1.040949248e+09
# HELP process_virtual_memory_max_bytes Maximum amount of virtual memory available in bytes.
# TYPE process_virtual_memory_max_bytes gauge
process_virtual_memory_max_bytes -1
# HELP promhttp_metric_handler_requests_in_flight Current number of scrapes being served.
# TYPE promhttp_metric_handler_requests_in_flight gauge
promhttp_metric_handler_requests_in_flight 1
# HELP promhttp_metric_handler_requests_total Total number of scrapes by HTTP status code.
# TYPE promhttp_metric_handler_requests_total counter
promhttp_metric_handler_requests_total{code="200"} 10248
promhttp_metric_handler_requests_total{code="500"} 0
promhttp_metric_handler_requests_total{code="503"} 0
# HELP rclone_bytes_transferred_total Total transferred bytes since the start of the Rclone process
# TYPE rclone_bytes_transferred_total counter
rclone_bytes_transferred_total 0
# HELP rclone_checked_files_total Number of checked files
# TYPE rclone_checked_files_total counter
rclone_checked_files_total 0
# HELP rclone_errors_total Number of errors thrown
# TYPE rclone_errors_total counter
rclone_errors_total 13
# HELP rclone_fatal_error Whether a fatal error has occurred
# TYPE rclone_fatal_error gauge
rclone_fatal_error 0
# HELP rclone_files_deleted_total Total number of files deleted
# TYPE rclone_files_deleted_total counter
rclone_files_deleted_total 0
# HELP rclone_files_renamed_total Total number of files renamed
# TYPE rclone_files_renamed_total counter
rclone_files_renamed_total 0
# HELP rclone_files_transferred_total Number of transferred files
# TYPE rclone_files_transferred_total counter
rclone_files_transferred_total 0
# HELP rclone_retry_error Whether there has been an error that will be retried
# TYPE rclone_retry_error gauge
rclone_retry_error 1
# HELP rclone_speed Average speed in bytes/sec since the start of the Rclone process
# TYPE rclone_speed gauge
rclone_speed 0
```

And here is a screenshot of the same instance's WebUI with the _real_ numbers:
![image](https://user-images.githubusercontent.com/14872/84528541-e2a4eb00-acad-11ea-8261-2142462da7d5.png)

As you can see, `rclone_errors_total` is 13, which is correct, but fields like `rclone_checked_files_total`, `rclone_bytes_transferred_total`, `rclone_speed`, `rclone_files_transferred_total`, etc, are all stuck at 0 incorrectly.

(Technically, I'm not 100% sure if the `process_*` variables are correct either, but they're at least not 0 and seem reasonable so I'm not doubting them at this time)

#### What is your rclone version (output from `rclone version`)

This screenshot and output is from v1.52.1 running in a Docker container, but I've also tried the `:beta` tag just to be sure, and it happens exactly the same way there too.

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)

Docker running on Arch Linux 64-bit.

####  Which cloud storage system are you using? (eg Google Drive)

The remote I was using to test this was a Google Drive with `crypt` on top.

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)

I'm running `rcd --enable-metrics` and then sending it `sync/sync` with `_async=true`.

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

Well, it's an `rc` call so the log from the call itself is not very interesting. I don't have the log for the `rcd` instance itself handy, but I had been monitoring it and it looked like a perfectly normal log output, plenty of "Copied (new)" entries for each file. Nothing on the WebUI or the log indicates rclone is doing anything wrong, the metrics just don't reflect reality.

**Comment by ncw** [comment]
Thanks for writing this up - excellent bug report :-)

@gary-kim you made the original patch for this - do you have time to take a look to see what is wrong?

**Comment by gary-kim** [comment]
Yep. I'll take a look

**Comment by ncw** [comment]
I've merged @gary-kim 's fix for this (thank you :-) this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.52.2

**Comment by apetresc** [comment]
I can confirm that it works properly in v1.52.2. Thank you @gary-kim! :)
